clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -291,12 +246,52 @@ public void completed(
           long offset,
           String message,
           Optional<LeaderFollowerStateType> leaderState) {
-        VeniceStoreVersionConfig config = configLoader.getStoreConfig(kafkaTopic);
-        config.setRestoreDataPartitions(false);
-        config.setRestoreMetadataPartition(false);
-        // Start partition consumption locally.
-        startConsumption(config, partition, leaderState);
+        if (isTopicPartitionIngesting(kafkaTopic, partition)) {
+          VeniceStoreVersionConfig config = configLoader.getStoreConfig(kafkaTopic);
+          config.setRestoreDataPartitions(false);
+          config.setRestoreMetadataPartition(false);
+          // Start partition consumption locally.
+          startConsumption(config, partition, leaderState);
+        } else {
+          LOGGER.error(
+              "Partition: {} of topic: {} is not assigned to this host, will not resume the ingestion on main process.",
+              partition,
+              kafkaTopic);
+        }
       }
     };
   }
+
+  private void executeCommandWithRetry(
+      String topicName,
+      int partition,
+      IngestionCommandType command,
+      Supplier<Boolean> remoteCommandSupplier,
+      Runnable localCommandRunnable) {
+    while (true) {
```

Comment:
It looks good. Nit, I tried to simplify it a bit more based on your change, ignore it if it doesn't make sense.
```
    while (true) {
      if (isTopicPartitionHostedInMainProcess(topicName, partition)) {
        LOGGER.info(
            "Executing command {} of topic: {}, partition: {} in main process process.",
            command,
            topicName,
            partition);
        localCommandRunnable.run();
        return;
      }
      LOGGER.info("Sending command {} of topic: {}, partition: {} to fork process.", command, topicName, partition);
      if (!remoteCommandSupplier.get()) {
       LOGGER.info(
          "Command {} rejected by remote ingestion process, will retry in {} ms.",
          command,
          RETRY_WAIT_TIME_IN_MS);
        if (!Utils.sleep(RETRY_WAIT_TIME_IN_MS)) {
          return;
        }
      }
    }
```

Reply:
Good point! Fixed.